src/Service/AbstractClient.php

```diff
@@ -6,40 +6,20 @@
 
 use Answear\WideEyesBundle\Exception\MalformedResponse;
 use Answear\WideEyesBundle\Exception\ServiceUnavailable;
-use Answear\WideEyesBundle\Request\GetSimilarRequest;
 use Answear\WideEyesBundle\Request\Request;
-use Answear\WideEyesBundle\Response\GetSimilarResponse;
 use GuzzleHttp\ClientInterface;
 use GuzzleHttp\Exception\GuzzleException;
 use Webmozart\Assert\Assert;
 
-class Client
+abstract class AbstractClient
 {
-    private const SEARCH_BY_ID_ENDPOINT = 'v4/SearchById';
+    protected const SEARCH_BY_ID_ENDPOINT = 'v4/SearchById';
+    protected const DETECT_AND_FEATURES_ENDPOINT = 'v4/DetectAndFeatures';
 
-    private ConfigProvider $configProvider;
-    private ?ClientInterface $guzzle;
+    protected ConfigProvider $configProvider;
+    protected ?ClientInterface $guzzle;
 
-    public function __construct(ConfigProvider $configProvider, ClientInterface $client = null)
```

Comment:
Ctor needs to stay here, otherwise there's no guarantee the properties will be initialized